### PR TITLE
Fix parsing of packages in pubspec.yaml that have no explicit version specification

### PR DESCRIPTION
**Pull request by banool**
Fixes https://github.com/fzyzcjy/flutter_rust_bridge/issues/826. As part of this I also make it print the errors in the failure case, so it's easier to see what's wrong with your spec.

I tested this against my own repo using the repro described in the above issue and the parsing step fixed it. I haven't fully tested the end to end flow though because I'm getting a different, likely unrelated issue related to llvm dynamic libraries.

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
  - I removed the version constraint for one of the more innocuous dependencies.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [x] (N/A) If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [ ] CI is passing.

### CODE REVIEW

File: frb_example/pure_dart/dart/pubspec.yaml
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/pubspec.yaml:18]: Hmm this does not sound very good in the final code :/

**Review by fzyzcjy — COMMENTED** [review]
LGTM except for one nit and wait for CI :)

File: frb_example/pure_dart/dart/pubspec.yaml
Comment (banool) [reply frb_example/pure_dart/dart/pubspec.yaml:18]: Ideally we have one unspecified dep in order to test this, do you have one in mind? I could also add a random, simple dep to dev_dependencies

Comment (fzyzcjy) [reply frb_example/pure_dart/dart/pubspec.yaml:18]: Maybe add a (rust) unit test

Comment (banool) [reply frb_example/pure_dart/dart/pubspec.yaml:18]: Alright sounds good, lemme do that.

**Comment by banool** [comment]
Okay added the unit test.

**Review by fzyzcjy — APPROVED** [review]
LGTM once tests pass :)

**Comment by banool** [comment]
CI looks good!

**Comment by fzyzcjy** [comment]
🎉

**Pull request merged**
